frontend/src/utils/useUserInputExpiration.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * useUserInputExpiration.js
+ */
+
+import { useEffect } from 'react';
+
+export default function useUserInputExpiration(expirationFunction, expirationDelay) {
+    let timer;
+
+    const startTimer = () => {
+        timer = setTimeout(expirationFunction, expirationDelay);
+    }
+
+    const resetTimer = () => {
+        if (timer)
+            clearTimeout(timer);
+        startTimer()
+    };
+
+    useEffect(() => {
+        const events = [
+            'mousedown',
+            'click',
```

Comment:
Unnecessary to have both `mousedown` and `click`. `click` will fire for any device (desktop, tablet, phone).